packages/datetime/src/dateRangeInput.tsx

```diff
@@ -733,16 +742,19 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
     }
 
     private getSelectedRange = () => {
-        const momentDateRange = [this.state.selectedStart, this.state.selectedEnd] as MomentDateRange;
-        const [startDate, endDate] = momentDateRange.map((selectedBound?: moment.Moment) => {
+        const { selectedStart, selectedEnd } = this.state;
+
+        // this helper function checks if the provided boundary date *would* overlap the selected
+        // other boundary date. providing the already-selected start date simply tells us if we're
+        // currently in an overlapping state.
```

Comment:
we don't do sentence casing in comments AFAIK, unless it's a prop-description comment.